fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/security/Ed25519/ElectionPublicKey.java

```diff
@@ -0,0 +1,133 @@
+package com.github.dedis.popstellar.model.objects.security.Ed25519;
+
+import android.util.Log;
+
+import com.github.dedis.popstellar.model.objects.security.Base64URLData;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.security.SecureRandom;
+import java.util.Objects;
+
+import ch.epfl.dedis.lib.crypto.Ed25519;
+import ch.epfl.dedis.lib.crypto.Ed25519Point;
+import ch.epfl.dedis.lib.crypto.Ed25519Scalar;
+import ch.epfl.dedis.lib.crypto.Point;
+import ch.epfl.dedis.lib.crypto.Scalar;
+import ch.epfl.dedis.lib.exception.CothorityCryptoException;
+import io.reactivex.annotations.NonNull;
+
+/**
+ * Represents a private key meant for El-Gam
+ */
+public class ElectionPublicKey {
+
+    // Point is generate with given public key
+    private final Point publicKey;
+
+    public ElectionPublicKey(@NonNull Base64URLData base64publicKey) {
+        try {
+            publicKey = new Ed25519Point(base64publicKey.getData());
+        } catch (CothorityCryptoException e) {
+            throw new IllegalArgumentException("Could not create the point for elliptic curve, please provide another key");
+        }
+    }
+
+    @Override
+    public String toString() {
+        return publicKey.toString();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        ElectionPublicKey that = (ElectionPublicKey) o;
+        return that.getPublicKey().equals(getPublicKey());
+    }
+
+    /**
+     * @return the string encoded public key in Base64 format
+     */
+    public String encodeToBase64() {
+        Base64URLData encodedKey = new Base64URLData(publicKey.toBytes());
+        return encodedKey.getEncoded();
+    }
+
+    /**
+     * @return the public key in Base64 format (not encoded)
+     */
+    public Base64URLData toBase64() {
+        return new Base64URLData(publicKey.toBytes());
+    }
+
+    public Point getPublicKey() {
+        return publicKey;
+    }
+
+    /**
+     * Added this hash behavior by default
+     */
+    @Override
+    public int hashCode() {
+        return java.util.Objects.hash(getPublicKey().toString());
+    }
+
+    /**
+     * Encrypts with ElGamal under Ed25519 curve
+     *
+     * @param message message a 2 byte integer corresponding to the chosen vote
+     * @return Returns 2 32 byte strings which should be appended
+     * together and base64 encoded. For clarification, the first 32 bytes is point K and the second 32 bytes is point C.
+     */
+    public String encrypt(@NonNull byte[] message) {
+        if (message.length > 29) {
```

Comment:
why 29?

Reply:
I found this number on the library, if you  have higher number, it will not properly embed.